### PR TITLE
Added support for a minimum configurable matching burst count

### PR DESCRIPTION
**Pull request by riverma**
## Purpose
- Support for a minimum matching burst count
## Proposed Changes
- [ADD] New command-line argument to support
- [CHANGE] Filtering logic to support new argument
- [CHANGE] README directions for filtering

## Issues
- N/A
## Testing
- Tested successfully with the below case:
```
python dswx_s1_validator.py --start "2023-12-05T01:00:00Z" --end "2023-12-05T03:59:59Z" --db MGRS_tile_collection_v0.3.sqlite --threshold 50

Total granules: 2316
Querying CMR for time range 2023-12-05T01:00:00Z to 2023-12-05T03:59:59Z.

Fetching granules: 100%|█████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 2316/2316 [00:04<00:00, 468.24it/s]

Granule fetching complete.

Calculating coverage: 100%|███████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 14687/14687 [00:13<00:00, 1122.64it/s]

MGRS Set IDs covered: 34
MGRS Set ID      Coverage Percentage    Matching Burst Count    Total Burst Count
MS_166_30                      70                         28                   40
MS_166_31                     100                         40                   40
MS_166_32                     100                         40                   40
MS_166_33                     100                         40                   40
MS_166_34                     100                         40                   40
MS_166_35                     100                         40                   40
MS_166_36                     100                         41                   41
MS_166_37                     100                         41                   41
MS_166_38                     100                         42                   42
MS_166_39                     100                         42                   42
MS_166_69                      90.24                      37                   41
MS_166_70                     100                         40                   40
MS_166_71                     100                         40                   40
MS_166_72                     100                         39                   39
MS_166_73                     100                         40                   40
MS_166_74                     100                         40                   40
MS_166_75                     100                         40                   40
MS_166_76                     100                         41                   41
MS_166_77                     100                         41                   41
MS_166_78                     100                         41                   41
MS_166_79                     100                         40                   40
MS_166_80                     100                         40                   40
MS_166_81                      50                         20                   40
MS_166_88                     100                         40                   40
MS_166_89                      90                         36                   40
MS_167_68                     100                         41                   41
MS_167_69                     100                         40                   40
MS_167_70                     100                         40                   40
MS_167_71                     100                         39                   39
MS_167_72                     100                         40                   40
MS_167_73                     100                         40                   40
MS_167_74                     100                         39                   39
MS_167_75                     100                         41                   41
MS_167_77                     100                         41                   41
```

Then:

```
python dswx_s1_validator.py --start "2023-12-05T01:00:00Z" --end "2023-12-05T03:59:59Z" --db MGRS_tile_collection_v0.3.sqlite --matching_burst_count 42
Total granules: 2316
Querying CMR for time range 2023-12-05T01:00:00Z to 2023-12-05T03:59:59Z.

Fetching granules: 100%|█████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 2316/2316 [00:03<00:00, 730.84it/s]

Granule fetching complete.

Calculating coverage: 100%|███████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 14687/14687 [00:11<00:00, 1237.32it/s]

MGRS Set IDs covered: 2
MGRS Set ID      Coverage Percentage    Matching Burst Count    Total Burst Count
MS_166_38                        100                      42                   42
MS_166_39                        100                      42                   42
```

### CODE REVIEW

File: report/dswx-s1-validator/dswx_s1_validator.py [hunk 1]
```diff
@@ -247,6 +247,7 @@ def get_burst_ids_from_query(start, end, timestamp, db):
     parser.add_argument("--db", required=True, help="Path to the SQLite database file")
     parser.add_argument("--file", required=False, help="Optional file path containing granule IDs")
     parser.add_argument("--threshold", required=False, help="Completion threshold minimum to filter results by (percentage format - leave out the % sign)")
+    parser.add_argument("--matching_burst_count", required=False, help="Matching burst count to filter results by. Typically 4 or more is advised.")
```
Comment (sjlewis-jpl) [inline report/dswx-s1-validator/dswx_s1_validator.py:250]: I think we should add a note here about what happens when providing both `--threshold` and `--matching-burst-count`, like in the README.

**Review by sjlewis-jpl — CHANGES_REQUESTED** [review]
Nice and tidy, well done.  One change to request in the help text at the command line.  I made a note in an inline comment.  Just to mention there what happens when both filtering options are passed to the script.

**Review by hhlee445 — APPROVED** [review]

File: report/dswx-s1-validator/dswx_s1_validator.py (hunk 1 above)
Comment (riverma) [inline report/dswx-s1-validator/dswx_s1_validator.py:250]:
```suggestion
    parser.add_argument("--matching_burst_count", required=False, help="Matching burst count to filter results by. Typically four or more is advised. Using this with the --threshold flag makes this flag inactive (only one of '--threshold' or '--matching_burst_count' may be used)")
```

Comment (riverma) [reply report/dswx-s1-validator/dswx_s1_validator.py]: Thanks for the suggestion @sjlewis-jpl. Took your advice and made an update to reflect. Thanks for your review!

**Pull request merged**